File: gluecodium/src/test/resources/smoke/generic_types/output/android/com/example/smoke/GenericTypesWithGenericTypes.java
/*
 *

 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
import java.util.List;
import java.util.Map;
import java.util.Set;
public final class GenericTypesWithGenericTypes extends NativeBase {
    /**
     * For internal use only.
     * @hidden
     * @param nativeHandle The SDK nativeHandle instance.
     * @param dummy The SDK dummy instance.
     */
    protected GenericTypesWithGenericTypes(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @NonNull
    public native List<List<Integer>> methodWithListOfLists(@NonNull final List<List<Integer>> input);
    @NonNull
    public native Map<Map<Integer, Boolean>, Boolean> methodWithMapOfMaps(@NonNull final Map<Integer, Map<Integer, Boolean>> input);
    @NonNull
    public native Set<Set<Integer>> methodWithSetOfSets(@NonNull final Set<Set<Integer>> input);
    @NonNull
    public native Map<Integer, List<Integer>> methodWithListAndMap(@NonNull final List<Map<Integer, Boolean>> input);
    @NonNull
    public native Set<List<Integer>> methodWithListAndSet(@NonNull final List<Set<Integer>> input);
    @NonNull
    public native Set<Map<Integer, Boolean>> methodWithMapAndSet(@NonNull final Map<Integer, Set<Integer>> input);
    @NonNull
    public native Map<List<Integer>, Boolean> methodWithMapGenericKeys(@NonNull final Map<Set<Integer>, Boolean> input);
}

File: gluecodium/src/test/resources/smoke/generic_types/output/android-kotlin/com/example/smoke/GenericTypesWithGenericTypes.kt
/*

 *
 */

@file:JvmName("GenericTypesWithGenericTypesExtensions")


package com.example.smoke

import com.example.NativeBase

class GenericTypesWithGenericTypes : NativeBase {



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}





    external fun methodWithListOfLists(input: List<List<Int>>) : List<List<Int>>


    external fun methodWithMapOfMaps(input: Map<Int, Map<Int, Boolean>>) : Map<Map<Int, Boolean>, Boolean>


    external fun methodWithSetOfSets(input: Set<Set<Int>>) : Set<Set<Int>>


    external fun methodWithListAndMap(input: List<Map<Int, Boolean>>) : Map<Int, List<Int>>


    external fun methodWithListAndSet(input: List<Set<Int>>) : Set<List<Int>>


    external fun methodWithMapAndSet(input: Map<Int, Set<Int>>) : Set<Map<Int, Boolean>>


    external fun methodWithMapGenericKeys(input: Map<Set<Int>, Boolean>) : Map<List<Int>, Boolean>




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}


